src/separations.h

```diff
@@ -0,0 +1,131 @@
+#ifndef CYCAMORE_SRC_SEPARATIONS_H_
+#define CYCAMORE_SRC_SEPARATIONS_H_
+
+#include "cyclus.h"
+
+namespace cycamore {
+
+cyclus::Material::Ptr SepMaterial(std::map<int, double> effs, cyclus::Material::Ptr mat);
+
+class Separations : public cyclus::Facility {
+ public:
+  Separations(cyclus::Context* ctx);
+  virtual ~Separations(){};
+
+  virtual void Tick();
+  virtual void Tock();
+  virtual void EnterNotify();
+
+  virtual void AcceptMatlTrades(const std::vector<std::pair<
+      cyclus::Trade<cyclus::Material>, cyclus::Material::Ptr> >& responses);
+
+  virtual std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
+  GetMatlRequests();
+
+  virtual std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr> GetMatlBids(
+      cyclus::CommodMap<cyclus::Material>::type& commod_requests);
+
+  virtual void GetMatlTrades(
+      const std::vector<cyclus::Trade<cyclus::Material> >& trades,
+      std::vector<std::pair<cyclus::Trade<cyclus::Material>,
+                            cyclus::Material::Ptr> >& responses);
+
+  #pragma cyclus clone
```

Comment:
It is placed where the pragma is.  In practice this isn't much of an issue
because nothing imports archetype headers.

On Fri, Apr 10, 2015 at 1:25 PM, Paul Wilson <notifications@github.com>
wrote:

> In src/separations.h
> <https://github.com/cyclus/cycamore/pull/309#discussion_r28168605>:
>
> > +
> > +  virtual void AcceptMatlTrades(const std::vector<std::pair<
> > +      cyclus::Trade<cyclus::Material>, cyclus::Material::Ptr> >& responses);
> > +
> > +  virtual std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
> > +  GetMatlRequests();
> > +
> > +  virtual std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr> GetMatlBids(
> > +      cyclus::CommodMap<cyclus::Material>::type& commod_requests);
> > +
> > +  virtual void GetMatlTrades(
> > +      const std::vector<cyclus::Trade<cyclus::Material> >& trades,
> > +      std::vector<std::pair<cyclus::Trade<cyclus::Material>,
> > +                            cyclus::Material::Ptr> >& responses);
> > +
> > +  #pragma cyclus clone
>
> where does the code-generated code get placed? at the location of the
> pragma? If so, this will bloat the header with code and we should consider
> the implications, if any?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cycamore/pull/309/files#r28168605>.
>